index.js

```diff
@@ -106,6 +118,11 @@
     return f(this.value);
   };
 
+  //  Just#concat :: m a -> m a
+  Just.prototype.concat = function(maybe) {
+    return maybe instanceof Just ? Just(this.value.concat(maybe.value)) : this;
+  };
+
```

Comment:
My understanding is that `Maybe a` is only a Semigroup if `a` is a Semigroup. `Number` is not a Semigroup, therefore `Maybe Number` is not a Semigroup.

I do understand your concern. It is unfortunate that `Just(42)` gives an object with a `concat` method *which should never be used*. I don't see a way around this. As for `a.concat(b)` where `a` and `b` are values of different Semigroups, we could certainly do type checking and throw a useful error, though this is not the approach we've taken in Ramda in most cases.